### DISCUSSION TITLE
Unattended Install of Open Watcom C++

### DISCUSSION THREAD

**Discussion by micahcochran** [open-watcom/open-watcom-v2#660] [Q&A] [answered]
I am trying to do an unattended (automated) install of Open Watcom C++ version 2.0 compiled on 2011-01-01.  Is there a way to do this?

What I am trying to do is set this for a CI process using a Ubuntu Linux instance that cross compile 16-bit DOS programs and run unit tests under DOSBox.

Or is there some other way to accomplish the installation?  Are there examples of other projects doing something similar?

**Comment by jmalak** [comment, marked as answer]
In [bld/setupgui/setup.txt](https://github.com/open-watcom/open-watcom-v2/blob/master/bld/setupgui/setup.txt) is some description of OW unattended installation (installer command line and configuration parameters).
It is not tested any way that some problem could exist.

Below are default values for some configuration parameters (fixed values), other parameters defaults depend on host OS etc. (non-fixed values).
ms=true
mm=true
mc=true
ml=true
mh=false
nlmtarg=false
rdoshost=false
rdostarg=false
samples=true
helpfiles=true
rtsource=false
3r=true
3s=true
cplusplus=true

**Comment by jmalak** [comment]
Next possibility is installation from OW build snapshot (ow-snapshot.tar.gz) by some script.
Snapshot contains "full" OW installation for all host and targets that you can copy only necessary sub-directories or files to your host environment and do appropriate environment variables setup by script.

**Sub-directories (not sub-tree) necessary for DOS host/target are:**
binw
h
h/sys
lib286
lib286/dos
lib386
lib386/dos

**Comment by micahcochran** [comment]
Thank you for the information and I appreciate the quick answer @jmalak.

I will keep in mind that unattended install is not be well tested.  If I run into any strange behavior, I will be sure to report it.
